server/databases.go

```diff
@@ -388,6 +427,81 @@ func (s *Service) DropRetentionPolicy(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// Measurements lists measurements within a database
+func (h *Service) Measurements(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	srcID, err := paramID("id", r)
+	if err != nil {
+		Error(w, http.StatusUnprocessableEntity, err.Error(), h.Logger)
+		return
+	}
+
+	limit, offset, err := validMeasurementQuery(r.URL.Query())
+	if err != nil {
+		Error(w, http.StatusUnprocessableEntity, err.Error(), h.Logger)
+		return
+	}
+
+	src, err := h.Store.Sources(ctx).Get(ctx, srcID)
+	if err != nil {
+		notFound(w, srcID, h.Logger)
+		return
+	}
+
+	db := h.Databases
+	if err = db.Connect(ctx, &src); err != nil {
+		msg := fmt.Sprintf("Unable to connect to source %d: %v", srcID, err)
+		Error(w, http.StatusBadRequest, msg, h.Logger)
+		return
+	}
+
+	dbID := httprouter.GetParamFromContext(ctx, "dbid")
+	measurements, err := db.GetMeasurements(ctx, dbID, limit, offset)
+	if err != nil {
+		msg := fmt.Sprintf("Unable to get measurements %d: %v", srcID, err)
+		Error(w, http.StatusBadRequest, msg, h.Logger)
+		return
+	}
+
+	res := measurementsResponse{
+		Measurements: measurements,
+		Links:        newMeasurementLinks(srcID, dbID, limit, offset),
+	}
+
+	encodeJSON(w, http.StatusOK, res, h.Logger)
+}
+
+func validMeasurementQuery(query url.Values) (limit, offset int, err error) {
+	limitParam := query.Get(limitQuery)
+	if limitParam == "" {
+		limit = 100
+	} else {
+		limit, err = strconv.Atoi(limitParam)
+		if err != nil {
+			return
+		}
+		if limit <= 0 {
```

Comment:
This check could be done once as deep as necessary to cover all call points

Reply:
Though it's also a valid argument that the API is responsible for coercing the request into a format that makes sense for the core to consume. So maybe do the check here, and free up `GetMeasurements` from doing that work?

Reply:
removed it from `newMeasurementsLinks`